third_party/move/move-bytecode-verifier/src/struct_defs.rs

```diff
@@ -91,8 +92,17 @@ impl<'a> StructDefGraphBuilder<'a> {
         let struct_def = StructDefinitionView::new(self.module, struct_def);
         // The fields iterator is an option in the case of native structs. Flatten makes an empty
         // iterator for that case
-        for field in struct_def.fields().into_iter().flatten() {
-            self.add_signature_token(neighbors, idx, field.signature_token(), false)?
+        let variant_count = struct_def.variant_count();
+        if variant_count > 0 {
+            for i in 0..variant_count {
+                for field in struct_def.fields_optional_variant(Some(i as VariantIndex)) {
+                    self.add_signature_token(neighbors, idx, field.signature_token(), false)?
+                }
+            }
+        } else {
+            for field in struct_def.fields_optional_variant(None) {
```

Comment:
nit: I think with this change, the `fields` method on `StructDefinitionView` is no longer used by anything in the codebase, so perhaps we can remove it?

Reply:
Yeah, the only concern is that whether this API may be used by external partners?

Reply:
I think any such use would likely be incorrect. But this could be done in a subsequent PR too.